api/services/RichText.js

```diff
@@ -1,54 +1,155 @@
+      el.getAttribute('data-user-id')
+    ) {
+      const newSpanElement = dom.createElement('span')
+
+      newSpanElement.className = 'mention'
+      newSpanElement.setAttribute('data-type', 'mention')
+      newSpanElement.setAttribute('data-id', el.getAttribute('data-user-id'))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
+
+      return
+    }
+
+    // Normalize legacy Topics
+    if (
+      el.getAttribute('data-entity-type') === '#mention' ||
+      (!el.getAttribute('href') && el.textContent[0] === '#')
+    ) {
+      const newSpanElement = dom.createElement('span')
+
+      newSpanElement.className = 'topic'
+      newSpanElement.setAttribute('data-type', 'topic')
+      newSpanElement.setAttribute('data-id', el.textContent?.slice(1))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
 
-  return $.html()
+      return
+    }
+  }, dom.querySelectorAll('a'))
+  
+  // Always sanitize on output, but only once and only here
+  const  santizedHTML = insane(
+    dom.querySelector('body').innerHTML,
+    TextHelpers.insaneOptions(providedInsaneOptions)
+  )
+
+  return santizedHTML
 }
 
 /*
-Returns a set of unique IDs for any mention members
-found in the provided HTML
 
-Used for generating notifications
+Prepares content for HTML Email delivery
+
+- Always make sure `processHTML` was ran first, this is done
+  in `Post#details()` and `Comment#text()`
+
+- Links will be generated with `/all` if a `groupSlug` is not passed
+
+- This same logic is handled dynamically on Web in `ClickCatcher`,
+  and on Mobile in the `HyloHTML` component.
+
 */
-export const getUserMentions = html => {
-  if (!html) return []
+export function qualifyLinks (processedHTML, groupSlug) {
+  const dom = getDOM(processedHTML)
+
+  // Convert Mention and Topic `span` elements to `a` elements
+  const convertSpansToAnchors = forEach(el => {
+    const anchorElement = dom.createElement('a')
+    let href = el.className === 'mention'
+      ? PathHelpers.mentionPath(el.getAttribute('data-id'), groupSlug)
+      : PathHelpers.topicPath(el.getAttribute('data-id'), groupSlug)
 
-  let $ = Cheerio.load(html)
+    for (const attr of el.attributes) {
+      anchorElement.setAttribute(attr.name, attr.value)
+    }
 
-  return _.uniq($('a[data-user-id]').map(function () {
-    return $(this).attr('data-user-id').toString()
-  }).get())
+    anchorElement.innerHTML = el.innerHTML
+    anchorElement.setAttribute('href', href)
+
+    el.parentNode.replaceChild(anchorElement, el)
+  })
+  convertSpansToAnchors(dom.querySelectorAll(
+    'span.topic, span.mention'
+  ))
+
+  forEach(el => {
+    const href = el.getAttribute('href')
+    if (href && !href.match(/^(https?|email):/)) {
```

Comment:
It's almost certainly a bug to be `Frontend.Route.prefix`'ing an `email:` URL 🤔 Tough obscured by the changes, this part was this way before. Can you seen any reason this would work? Shouldn't we just leave `email:` links alone?